fix: Colorize example tests

karlkfi opened the pull request:
Makes the `expectedOutputLine` and `expectedNotFound` tests succeed green and fail red, with re-printed expectation on failure, to make it easier to see problems at a glance when running locally.

karlkfi commented:
/test cli-utils-presubmit-master-e2e

flaky failure?

karlkfi commented:
Tests fixed.

karlkfi commented on examples/alphaTestExamples/MultipleServices.md:
```diff
@@ -47,20 +50,24 @@ curl -s -o "$BASE/mysql/#1.yaml" "https://raw.githubusercontent.com\
+  if ! grep -q "$@" "$OUTPUT/status"; then
+    echo -e "${RED}Error: output line not found${NC}"
+    echo -e "${RED}Expected: $@${NC}"
+    exit 1
```
Go tests have the concept of Error vs Fatal, where error continues and fatal does not. I'm not sure which tests in the examples should be errors and which should be fatal, so I assumed they were all fatal. Do you know the tests well enough to make the distinction for each one?

mortent replied:
I think failing on the first error within each md file probably make sense. The commands depend on each other, so continuing will likely just result in other errors.
Can we just run mdrip once for each file and continue even if one of them fails?

mortent replied:
This is something we can change quickly at any time if we change our minds, so I'm not super worried about it either way.

karlkfi replied:
> Can we just run mdrip once for each file and continue even if one of them fails?

Done.

karlkfi commented on examples/alphaTestExamples/MultipleServices.md:
```diff
@@ -47,20 +50,24 @@ curl -s -o "$BASE/mysql/#1.yaml" "https://raw.githubusercontent.com\
+    echo -e "${RED}Expected: $@${NC}"
+    exit 1
+  else
+    echo -e "${GREEN}Success: output line found${NC}"
```
If you have an idea of how to do that succinctly, I'm all ears. I decided not to re-print the entire output again. And I don't know how to easily get the command run or the line number... Got ideas?

karlkfi replied:
FWIW, with every test fatal, we can at least tell which code block caused the failure, if not the exact line.

mortent replied:
I think the only reasonable way to do this would be to print the text that was expected. But it might be that it clutters  up the output so it isn't much help.

haiyanmeng commented:
/lgtm

karlkfi commented:
Rebased and modified the test script to execute each file individually, so that the new exit on error behavior only affects the file being tested.

Please re-review.

karlkfi commented:
`pruneAndDelete.md` was using the `expectedNotFound` function from `helloapp.md`, causing test failure now that they exec in different mdrip contexts (I didn't know shell variables were shared between files).

Fixed by copying `expectedNotFound` to `pruneAndDelete.md`.

karlkfi commented:
Tests passed.

seans3 reviewed (approved):
I'll let others have look before merge.

mortent reviewed (commented):
/lgtm
